python/test/test_dataset.py

```diff
@@ -465,6 +465,26 @@ def test_Workspace2D_example(self):
                 spec[Data.Value, "sample1"] = np.zeros(1000)
                 spec[Data.Variance, "sample1"] = np.zeros(1000)
 
+    def test_zip(self):
+        d = Dataset()
+        d[Coord.SpectrumNumber] = ([Dim.Position], np.arange(1, 6))
+        d[Data.EventTofs, ""] = ([Dim.Position], (5,))
+        d[Data.EventPulseTimes, ""] = ([Dim.Position], (5,))
+        self.assertEqual(len(d[Data.EventTofs, ""].data), 5)
+        d[Data.EventTofs, ""].data[0].append(10)
+        d[Data.EventPulseTimes, ""].data[0].append(1000)
+        d[Data.EventTofs, ""].data[1].append(10)
+        d[Data.EventPulseTimes, ""].data[1].append(1000)
+        # Don't do this, there are no compatiblity checks:
+        #for el in zip(d[Data.EventTofs, ""].data, d[Data.EventPulseTimes, ""].data):
+        print("zip start")
```

Comment:
Not sure what you mean exactly by your comment here (above). Is that outdated? What you have below looks fine to me.

Reply:
If you just `zip` the underlying data on the Python side, you can end up combining two arrays with mismatching dimensions. It would be correct in this case, but if one of the variables has extra dimensions (or transposed dimensions) the result will be garbage.